sklearn/metrics/ranking.py

```diff
@@ -149,15 +197,30 @@ def average_precision_score(y_true, y_score, average="macro",
     sample_weight : array-like of shape = [n_samples], optional
         Sample weights.
 
+    interpolation : string ['linear' (default), 'step']
+        Determines the kind of interpolation used when computed AUC. If there
+        are many repeated scores, 'step' is recommended to avoid under- or
+        over-estimating the AUC. See `Roam Analytics blog post
```

Comment:
This doesn't seem a good source to me. I'm happy with citing the IR book, though.